lib/IDE/Refactoring.cpp

```diff
@@ -1667,6 +1667,366 @@ bool RefactoringActionConvertStringsConcatenationToInterpolation::performChange(
   return false;
 }
 
+struct ExpandableTernaryExprInfo {
```

Comment:
Can we make two sub-class of ExpandableTernaryExprInfo? Currently the logic for PatternBindingDecl and AssignExpr seems to be intertwined.